specs/models/Step.v1.json

```diff
@@ -0,0 +1,181 @@
+{
+  "title": "Step.v1",
+  "type": "object",
+  "properties": {
+    "options": {
+      "type": "object",
+      "properties": {
+        "sourceQuery": {
+          "type": "string",
+          "description": "Defines the items to be processed by the step; must be a cts.query or cts.uris statement if sourceQueryIsScript is false"
+        },
+        "sourceQueryIsScript": {
+          "type": "boolean",
+          "description": "Added in 5.3.0; if true, then sourceQuery can be any JavaScript statement that can be passed into xdmp.eval"
+        },
+        "constrainSourceQueryToJob": {
+          "type": "boolean",
+          "description": "If true, the query is applied to the documents that were created or modified in the same job that executes the step"
+        },
+        "provenanceGranularityLevel": {
+          "type": "string",
+          "description": "The granularity of the provenance tracking information: coarse (default) to store document-level provenance information only, fine to store document-level and property-level provenance information, or off to disable provenance tracking in future job runs. Applies only to mapping, matching, merging, mastering, and custom steps.",
+          "enum": [
+            "off",
+            "coarse",
+            "fine"
+          ]
+        },
+        "stepUpdate": {
```

Comment:
If the intent is to define a new Step schema that uses "options" strictly for things that make sense to override at runtime... you know, I'm wondering if "options" should be purely a runtime thing in the first place? i.e. put sourceQuery directly under Step, and a Step has no understanding of "options" in the first place. Then, some chunk of code can look at the options defined at runtime when a flow is run, and it'll use e.g. Object.assign to overwrite what's in the step definition. So e.g.:

```
options = {}; // provided by user at runtime
flow = getFlow();
stepId = flow.steps["1"].stepId;
step = getStep(stepId); // loads a step document from the database
step = applyOptionsToStep(step);
flow.steps["1"] = step;
```

That makes the data model much easier to design and understand, because now we don't need to think about "Which of these things would a user want to override at runtime?" If needed, the "applyOptionsToStep" function could prohibit certain things from being overridden.

Thoughts on dumping "options" completely from a Step?

Reply:
"options" for a Step(as they appear in a flow) really are "settings" based on current implementation right?

```
"steps": {
  "1": {
    "name": "dhsLoadData", 
    "description": "", 
    "options": {
        "sourceQuery": null, 
        "collections": ["default-ingestion"], 
        "loadData": {
            "name": "dhsLoadData"
        }, 
        "outputFormat": "json", 
        "targetDatabase": "data-hub-STAGING"
    }, 
```

Reply:
"Settings" is an arbitrary concept that appears in the UI but could change at any point. We don't want to model our data based how form dialogs are constructed in the UI.

Reply:
@rjrudin  The only scenario where I'm not certain about dropping options all together is the scenario of custom steps. Do we just pass in the entire step into the step function instead of options in that case? If not, I'm not certain how we'd determine what qualifies as "options" for the custom step.

Reply:
Each Step might need a different set of options. Command line options would apply it commonly across all steps right?

Reply:
The intent of options is to allow for some parts of a step to be overridden at runtime. But "some" is not clearly defined, and there are clear instances where what's in and what's not in "options" doesn't make sense:

- batchSize and threadCount are not in options, but those clearly would be overridden at runtime
- fileLocations/input-file-path is not in options, but is likely to be overridden at runtime
- Examples of properties in options that are extremely unlikely to be overridden at runtime: outputFormat, acceptsBatch, stepUpdate, mapping

My point is that "options" as defined in a step are meaningless today. We should discard it - it's an unnecessary level of nesting in a step that provides no meaning, as demonstrated above. 

Runtime options make sense though. This should be an object that can overwrite any part of a step, though the function that applies options to a step can always prohibit certain things from being overridden as needed. For example, we likely don't want to allow for name/stepDefinitionName/stepDefinitionType to be overridden.

Reply:
removed "options" from the "Step.v1" model. Left "options" as is in the Flow schema.